### PR TITLE
#34 adds functionality to represent data in stem and leaf plot

### PR DESCRIPTION
**Pull request by niraj-rajbhandari**
This pull request consists of functionality to
- get data points from the user
- generate a stem and leaf plot to represent the data provided by the user

Please review the code.

### CODE REVIEW

**Comment by niraj-rajbhandari** [comment]
This pull request consists of functionality to

```
- get data points from the user
- generate a stem and leaf plot to represent the data provided by the user
```

Please review the code.

File: src/com/lftechnology/java/training/niraj/stemleaf/StemLeafPlot.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+/**
+ * 
+ * Plots stem and leaf
+ * 
+ * @author Niraj Rajbhandari <nirajrajbhandari@lftechnology.com>
+ *
+ */
+public class StemLeafPlot {
+	private final static Logger LOGGER = Logger.getLogger(SumOfMultiples.class
```
Comment (bhuwang) [inline src/com/lftechnology/java/training/niraj/stemleaf/StemLeafPlot.java]: MINOR Reorder the modifiers to comply with the Java Language Specification. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AModifiersOrderCheck)

File: src/com/lftechnology/java/training/niraj/stemleaf/StemLeafPlot.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+	 * @param scanner
+	 *            {@link Scanner}
+	 * @return {@link Integer[]}
+	 * @author Niraj Rajbhandari <nirajrajbhandari@lftechnology.com>
+	 */
+	private static int[] getDataPoints(int dataPointLength, Scanner scanner) {
+		int[] dataPoints = new int[dataPointLength];
+		int dataPoint;
+		for (int i = 0; i < dataPointLength; i++) {
+			System.out.println("datapoint[" + i + "] :");
```
Comment (bhuwang) [inline src/com/lftechnology/java/training/niraj/stemleaf/StemLeafPlot.java]: MAJOR Replace this usage of System.out or System.err by a logger. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS106)

File: src/com/lftechnology/java/training/niraj/stemleaf/StemLeafPlot.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+					scanner.next();
+				} else {
+					break;
+				}
+			}
+			int[] dataPoints = getDataPoints(dataPointLength, scanner);
+			StemLeaf stemLeaf = new StemLeaf(dataPoints);
+			stemLeaf.plotStemLeaf();
+
+		} catch (InputMismatchException e) {
```
Comment (bhuwang) [inline src/com/lftechnology/java/training/niraj/stemleaf/StemLeafPlot.java]: CRITICAL Either log or rethrow this exception. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)

File: src/com/lftechnology/java/training/niraj/stemleaf/StemLeaf.java [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+		List<Integer> leafList;
+
+		System.out.format("%s\n%s\n", "Sorted Array :",
+				Arrays.toString(dataPoints));
+		System.out.format("%s\n%1s%23s%15s\n%s", "Stem and leaf Display",
+				"Frequency", "Stem", "Leaf",
+				"====================================================");
+		for (Entry<Integer, List<Integer>> entry : stemLeafInfo.entrySet()) {
+			leafList = entry.getValue();
+			System.out.format("\n%1d%30d%15s", leafList.size(), entry.getKey(),
```
Comment (bhuwang) [inline src/com/lftechnology/java/training/niraj/stemleaf/StemLeaf.java]: CRITICAL %n should be used in place of \n to produce the platform-specific line separator. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2275)
MAJOR Replace this usage of System.out or System.err by a logger. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS106)

File: src/com/lftechnology/java/training/niraj/stemleaf/StemLeaf.java [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+	 * Display the stem and leaf plot
+	 * 
+	 * @author Niraj Rajbhandari <nirajrajbhandari@lftechnology.com>
+	 */
+	private void displayStemLeafPlot() {
+		List<Integer> leafList;
+
+		System.out.format("%s\n%s\n", "Sorted Array :",
+				Arrays.toString(dataPoints));
+		System.out.format("%s\n%1s%23s%15s\n%s", "Stem and leaf Display",
```
Comment (bhuwang) [inline src/com/lftechnology/java/training/niraj/stemleaf/StemLeaf.java]: MAJOR Replace this usage of System.out or System.err by a logger. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS106)
CRITICAL %n should be used in place of \n to produce the platform-specific line separator. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2275)

File: src/com/lftechnology/java/training/niraj/stemleaf/StemLeaf.java [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+
+	/**
+	 * Display the stem and leaf plot
+	 * 
+	 * @author Niraj Rajbhandari <nirajrajbhandari@lftechnology.com>
+	 */
+	private void displayStemLeafPlot() {
+		List<Integer> leafList;
+
+		System.out.format("%s\n%s\n", "Sorted Array :",
```
Comment (bhuwang) [inline src/com/lftechnology/java/training/niraj/stemleaf/StemLeaf.java]: MAJOR Replace this usage of System.out or System.err by a logger. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS106)
CRITICAL %n should be used in place of \n to produce the platform-specific line separator. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2275)

File: src/com/lftechnology/java/training/niraj/stemleaf/StemLeaf.java [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+	 *            {@link Integer[]}
+	 * @author Niraj Rajbhandari <nirajrajbahndari@lftechnology.com>
+	 */
+	public void setStemLeafInfo(int[] dataPoints) {
+		int quotient;
+		int remainder;
+		float divisionResult;
+		for (int i = 0; i < dataPoints.length; i++) {
+			remainder = dataPoints[i] % DIVIDER;
+			divisionResult = dataPoints[i] / DIVIDER;
```
Comment (bhuwang) [inline src/com/lftechnology/java/training/niraj/stemleaf/StemLeaf.java]: MAJOR Cast one of the operands of this division operation to a "float". [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2184)

**Comment by bhuwang** [comment]
SonarQube analysis reported 12 issues:
* CRITICAL 4 critical
* MAJOR 5 major
* MINOR 3 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Replace all tab characters in this file by sequences of white-spaces.](https://github.com/bhuwang/java_training/blob/7f2a79a66af40e0aa62293a2ff62873b1dcbc973/src/com/lftechnology/java/training/niraj/stemleaf/StemLeaf.java) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00105)
* MINOR [Replace all tab characters in this file by sequences of white-spaces.](https://github.com/bhuwang/java_training/blob/7f2a79a66af40e0aa62293a2ff62873b1dcbc973/src/com/lftechnology/java/training/niraj/stemleaf/StemLeafPlot.java) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00105)

**Comment by bhuwang** [comment]
SonarQube analysis reported 3 issues:
* CRITICAL 1 critical
* MINOR 2 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Replace all tab characters in this file by sequences of white-spaces.](https://github.com/bhuwang/java_training/blob/f73fd503bf02964af1a81310a6ead4d4866a0523/src/com/lftechnology/java/training/niraj/stemleaf/StemLeaf.java) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00105)
* MINOR [Replace all tab characters in this file by sequences of white-spaces.](https://github.com/bhuwang/java_training/blob/f73fd503bf02964af1a81310a6ead4d4866a0523/src/com/lftechnology/java/training/niraj/stemleaf/StemLeafPlot.java) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00105)

**Pull request merged**